### PR TITLE
Support passing config as paramater to deliver/2

### LINKED ISSUE #46: optionally passing config as parameter when sending email
Example: Mailgun. It requires you to use the domain you are sending from in API calls. Some sites use more than one domain.

If you have an app that uses several domains, it would be nice if you could pass the domain as part of the parameters to the sending function.

### PR DESCRIPTION
**Pull request by stevedomin**
For some providers it's useful to be able to override the Mailer
config on a per-email basis. For instance, Mailgun requires you
to specify the domain you are sending from. If you are building
an app that's supporting several domains then you can use that
feature to pass the domain when making the call to deliver/2.

Fixes #46

### CODE REVIEW

**Comment by stevedomin** [comment]
@lau is that what you had in mind?

File: lib/swoosh/mailer.ex [hunk 1]
```diff
@@ -56,19 +56,28 @@ defmodule Swoosh.Mailer do
...
-      def deliver(%Swoosh.Email{subject: nil}), do: raise ArgumentError, "expected \"subject\" to be set"
-      def deliver(%Swoosh.Email{html_body: nil, text_body: nil}) do
+      def deliver(email, config \\ [])
+      def deliver(%Swoosh.Email{from: nil}, _config), do:
+        raise ArgumentError, "expected \"from\" to be set"
+      def deliver(%Swoosh.Email{from: {_name, nil}}, _config), do:
+        raise ArgumentError, "expected \"from\" address to be set"
+      def deliver(%Swoosh.Email{to: nil}, _config), do:
+        raise ArgumentError, "expected \"to\" to be set"
+      def deliver(%Swoosh.Email{to: []}, _config), do:
```
Comment (barisbalic) [inline lib/swoosh/mailer.ex]: Mixing single/multiline?  Stick to single here if poss.

**Comment by barisbalic** [comment]
Aside from comment :+1:

File: lib/swoosh/mailer.ex (hunk 1 above)
Comment (stevedomin) [inline lib/swoosh/mailer.ex]: It's really unreadable, happy to explore other options but this is by far the most readable

**Comment by lau** [comment]
Yeah it looks like it. Would be nice with an example of passing custom run config too. Here: https://github.com/swoosh/swoosh/blob/d4b49ab4f633f72da624e7d96678308ae4709eab/lib/swoosh/mailer.ex#L39

Once this is released I will probably begin looking at switching some production apps I have running from the :mailgun package to Swoosh.

**Comment by barisbalic** [comment]
@lau that's pretty cool.  Keep us posted on how that goes and if anything obvious crops up.

File: lib/swoosh/mailer.ex (hunk 1 above)
Comment (barisbalic) [inline lib/swoosh/mailer.ex]: I find it pretty readable, but for now we can crack on.  I'm not happy about mixing the one line/multiline syntax though, `def .. do .. end`

**Comment by stevedomin** [comment]
@lau thanks, good point, I added an example.

If you need any help or needs any bugs fixed you know where to find us

**Comment by stevedomin** [comment]
@lau fyi we've released swoosh 0.2: https://hex.pm/packages/swoosh

**Comment by lau** [comment]
Cool. I'll check it out.

**Pull request merged**